daceml/onnx/op_implementations/fpga_implementations.py

```diff
@@ -886,6 +886,1132 @@ def make_compute(sdfg, state, vec_width=1):
+            load_test = f"(k % {filter_hx * filter_hy + 1} == 0)"
+
+            # compute the input channel index for DRAM using the corrected k for buffering phase
+            channel_load = f"int_floor({k_load}, ({filter_hx * filter_hy}))"
+
+            # Translate from matrix location to pixel stored in (b,c,x,y)
+            m = f"(m0 * {vec_width})"
+            matrix_col = f"(tm*{T} + {m})"  # untiled matrix column access
+            out_y = f"int_floor({matrix_col}, {output_size_x})"  # y position in output
+            out_y_cpp = f"{matrix_col} / {output_size_x}"  # y position in output
+            out_x = f"{matrix_col} % {output_size_x}"  # x position in output
+
+            # Filter offsets
+            filter_off_y_img = f"int_floor(({k_img} % ({filter_hx} * {filter_hy})), {filter_hx})"
+            filter_off_y_cpp_img = f"({k_img_cpp} % ({filter_hx} * {filter_hy})) / {filter_hx}"
+            filter_off_x_img = f"({k_img} % ({filter_hx} * {filter_hy})) % {filter_hx}"
+            filter_off_x_cpp_img = f"({k_img_cpp} % ({filter_hx} * {filter_hy})) % {filter_hx}"
+
+            # Actual data element to access
+            access_y_img = f"({out_y}) + {filter_off_y_img}"
+            access_y_cpp_img = f"({out_y_cpp}) + {filter_off_y_cpp_img}"
+            access_x_img = f"({out_x}) + {filter_off_x_img}"
+            access_x_cpp_img = f"({out_x}) + {filter_off_x_cpp_img}"
+
+            # Compute coverage of a tile
+            tile_input_coverage = f"(int_floor({T}, {output_size_x}) * {input_size_x * vec_width_in})"
+            tile_input_coverage_cpp = f"(({T}/ {output_size_x}) * {input_size_x * vec_width_in})"
+
+            # Compute accessed pixel index in the buffering phase
+            # During buffering we can read aligned and sequentially from memory using the
+            # loop variable `m0` but we have to know how much of the image has already
+            # been covered by previous tiles.
+            accessed_pixel = f"(tm*int_floor({tile_input_coverage}, {vec_width_in}) + m0)"
+            accessed_pixel_cpp = f"(tm*({tile_input_coverage_cpp}/ {vec_width_in}) + m0)"
+
+            # Compute final access on the (b,c,x,y) data layout considering padding.
+            access_load = f"[b, {channel_load},  min({input_size_y - 1 + (2 * padding)}, int_floor({accessed_pixel}, {input_size_x})) - {padding}, {accessed_pixel} % {input_size_x}]"
+
+            # access within the local BRAM buffer
+            store_local_index_img = f"(({access_y_img}) * {input_size_x * vec_width_in} + ({access_x_img}) - {padding}) - ({tile_input_coverage} * tm)"
+
+            # If we are out-of bound, use a dummy value
+            new_sdfg.add_array("B_dummy",
+                               dtype=base_type,
+                               shape=[1],
+                               transient=True,
+                               storage=dace.dtypes.StorageType.FPGA_Registers)
+            b_dummy = state.add_access("B_dummy")
```

Comment:
Why does this need to be a data container? If it ever holds something useful, it's not a dummy. If not, why not just inject a constant?

Reply:
Good point yes. I have seen Tiziano do it in several places. @TizianoDeMatteis any reasons why you chose to make this a register value and not directly a constant?

Reply:
Yeah, it makes more sense to use a constant (if there was a reason, I forgot what it was about)

Reply:
Ok, changed it to be a constant 👍

Guess why it could have made sense to make it a register is to support different constants for padding. But I think I have never really seen something different than 0 in common applications. So this should definitely be fine.

Reply:
If we _did_ want a register for different values, it should not be called "dummy" anyway ;-) Remove for now, reinstate only if there's a use case!